### DISCUSSION TITLE
possible inconsistency in units of cell and positions

### DISCUSSION THREAD

**Discussion by gabriele16** [MarsalekGroup/aml#12] [Q&A] [answered]
Thanks for making the code available, really nice. Is there a possible inconsistency in the way that units are parsed for the cell and for the positions when using read_frames_cp2k? The function takes in the positions in Angstroms but the cell in bohr. I might have missed if this is already documented in the code. Anyway, changing to Angstrom as default for the cell in cp2k might help others not to encounter the same problem.

**Comment by OndrejMarsalek** [comment, marked as answer]
There is a method to the madness, but it should be better documented, so we will do that - thank you for bringing it up.

The idea is that all data within AML itself is in atomic units. Units in external files are determined either by the format itself, by convention (position data in XYZ, for example), or by whatever a specific program does. As `read_frames_cp2k` gets the name of an XYZ file produced by CP2K, that has to be in angstrom, but internally, the position data is converted to atomic units after being read. If there is a cell provided to the function by hand, that has to be in atomic units, as it is AML data directly in Python. You can easily grab the factor from `aml.constants` and (as an example for a 10 A cubic cell) say something like the following so that you can use values in angstrom in your code for clarity.
```
[...]
L = 10.0
cell = L * aml.constants.angstrom * np.eye(3)
read_frames_cp2k(fn_positions, cell=cell, fn_forces=fn_forces)
```

**Reply by gabriele16** [reply]
Thanks a lot!
